packages/codegen/src/EntityDbMetadata.ts

```diff
@@ -731,6 +739,12 @@ function superstructType(s: string): Import {
   return Import.from(`${pascalCase(symbol)}@${path.join("@")}`);
 }
 
+function zodSchemaType(s: string): Code {
+  const schema = Import.from(s);
+  const zod = Import.from(`z@zod`);
+  return code`${zod}.infer<typeof ${schema}>`;
```

Comment:
I saw your PR description re `z.output` / `z.input`--the dual input/output types are neat.

Just curious, but afaiu isn't `z.infer` just an alias for `z.output`? If so, could we use `z.output` here just for simplicity/consistency with the getter?

Reply:
Yup! I'll change it. Bit silly really, used infer as the default as I didn't want to pass judgement on the other places fieldType is used.

Reply:
Ah yep, totally get using `z.infer` at first b/c yeah I'm anticipating eventually needing to teach more places about the input-vs-output type difference.

I.e. as-is you've fixed the setter, but places like `em.create`, `new Author(em, ...)`, and `newAuthor(em, ...)` are still using the `AuthorOpts`/output type.

But, if we just flipped the `Opts`/`fieldType` over to the input type, I'm sure we'd have an equal/non-zero number of places that would still want the output type.

So, dunno, probably not something to tackle with this PR, but yep I imagine we'll eventually have to track all/most of the `fieldType`-for-jsonb-usages down and teach them about input-vs-output type...

Reply:
With those uses you've listed above, I'm erring on using `z.input` as the default.

I'd perhaps propose in a follow up the idea of allowing us to optionally split field type.

```ts
type FieldTypeType = PrimitiveTypescriptType | Import | Code;

type FieldType = fieldTypeType | { input: fieldTypeType, output: fieldTypeType }
```

and then usages wise we can introduce

```
function getInputFieldType(fieldType: FieldType) {
	return 'input' in fieldType ? fieldType.input : fieldType;
}

function getOutputFieldType(fieldType: FieldType) {
	return 'input' in fieldType ? fieldType.input : fieldType;
}
```

This would allow us to define intent at the point of usage in codegen and keep the field logic in EntityDbMetadata.